cmake/GinkgoConfig.cmake.in

```diff
@@ -147,10 +152,14 @@ endif()
 # NOTE: we do not export benchmarks, examples, tests or devel tools
 #     so `third_party` libraries are currently unneeded.
 
-# propagate CUDA_HOST_COMPILER if Ginkgo was built with CUDA
-if (GINKGO_BUILD_CUDA AND GINKGO_CUDA_HOST_COMPILER AND NOT CMAKE_CUDA_HOST_COMPILER)
-    message(STATUS "Ginkgo: Setting CUDA host compiler to ${GINKGO_CXX_COMPILER}")
-    set(CMAKE_CUDA_HOST_COMPILER "${GINKGO_CXX_COMPILER}" CACHE STRING "" FORCE)
+# propagate CUDA_HOST_COMPILER if Ginkgo needed
+if (GINKGO_BUILD_CUDA OR (GINKGO_BUILD_HIP
+    AND GINKGO_HIP_PLATFORM MATCHES "${GINKGO_HIP_PLATFORM_NVIDIA_REGEX}"))
+    if (GINKGO_CUDA_HOST_COMPILER AND NOT CMAKE_CUDA_HOST_COMPILER
+        AND EXISTS "${GINKGO_CUDA_HOST_COMPILER}")
```

Comment:
Good idea! Especially if Ginkgo got packaged in binary form on some system with a home-built gcc, this might bite us otherwise.